tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs

```diff
@@ -364,12 +445,7 @@ public static void BubbleSortArray()
         [InlineData(1000)]
         public static void TestSpanConstructorByte(int length)
         {
-            var array = new byte[length];
-            Span<byte> span;
-            foreach (var iteration in Benchmark.Iterations)
-                using (iteration.StartMeasurement())
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                        span = new Span<byte>(array);
+            InvokeTestSpanConstructor<byte>(length);
```

Comment:
IMO the different overloads are easier to read since they don't need the reflection magic, and they also have the benefit you mentioned in your edit.  I don't mind changing it if you disagree, though.